src/Microsoft.DotNet.Wpf/src/WindowsBase/System/Windows/Input/KeyConverter.cs

```diff
@@ -4,246 +4,326 @@
-                if (key != null)
-                {
-                    return ((Key)key);
-                }
-                else
-                {
-                    throw new NotSupportedException(SR.Format(SR.Unsupported_Key, fullName));
-                }
-            }
-            throw GetConvertFromException(source);
+            if (source is not string stringSource)
+                throw GetConvertFromException(source);
+
+            ReadOnlySpan<char> fullName = stringSource.AsSpan().Trim();
+            return GetKeyFromString(fullName);
         }
 
         /// <summary>
-        /// ConvertTo()
+        /// Converts a <paramref name="value"/> of <see cref="Key"/> to its <see langword="string"/> represensation.
         /// </summary>
-        /// <param name="context"></param>
-        /// <param name="culture"></param>
-        /// <param name="value"></param>
-        /// <param name="destinationType"></param>
-        /// <returns></returns>
-        /// <ExternalAPI/> 
+        /// <param name="context">Serialization Context</param>
+        /// <param name="culture">Culture Info</param>
+        /// <param name="value">Key value </param>
+        /// <param name="destinationType">Type to Convert</param>
+        /// <returns>A <see langword="string"/> representing the <see cref="Key"/> specified by <paramref name="value"/>.</returns>
         public override object ConvertTo(ITypeDescriptorContext context, CultureInfo culture, object value, Type destinationType)
         {
             ArgumentNullException.ThrowIfNull(destinationType);
 
-            if (destinationType == typeof(string) && value != null)
+            if (value is null || destinationType != typeof(string))
+                throw GetConvertToException(value, destinationType);
+
+            Key key = (Key)value;
+            return key switch
             {
-                Key key = (Key)value;
-                if (key == Key.None)
-                {
-                    return String.Empty;
-                }
-
-                if (key >= Key.D0 && key <= Key.D9)
-                {
-                    return Char.ToString((char)(int)(key - Key.D0 + '0'));
-                }
-
-                if (key >= Key.A && key <= Key.Z)
-                {
-                    return Char.ToString((char)(int)(key - Key.A + 'A'));
-                }
-
-                String strKey = MatchKey(key, culture);
-                if (strKey != null)
-                {
-                    return strKey;
-                }
-            }
-            throw GetConvertToException(value, destinationType);
+                Key.None => string.Empty,
+                // This is a fast path for common keys before resort to Enum<Key>.ToString()
+                >= Key.D0 and <= Key.D9 => char.ToString((char)(key - Key.D0 + '0')),
+                >= Key.A and <= Key.Z => char.ToString((char)(key - Key.A + 'A')),
```

Comment:
This could be sped up from 6ns to 2ns per invocation using static array (takes 10ns to create and 330B in memory)
```csharp
private static readonly string[] s_letters = ["A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z"];
private static readonly string[] s_digits = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"];
```
and using array-lookup instead
```
>= Key.D0 and <= Key.D9 => s_digits[key - Key.D0],
>= Key.A and <= Key.Z => s_letters[key - Key.A],
```

Opinions?

| Method            | source | Mean [ns] | Error [ns] | StdDev [ns] | Gen0   | Code Size [B] | Allocated [B] |
|------------------ |------- |----------:|-----------:|------------:|-------:|--------------:|--------------:|
| Original| A      |  5.796 ns |  0.1326 ns |   0.1108 ns | 0.0014 |         381 B |          24 B |
| PR__EDIT| A      |  2.281 ns |  0.0134 ns |   0.0119 ns |      - |         575 B |             - |

Reply:
> 5ns to 2ns per invocation [...] Opinions?

Because you asked for it: Smaller number always look more impressive when it comes to optimization. But if we speak about 3 nanoseconds more or less, I think we're at the level of micro optimization. Declaring additional arrays just to optimize two lines of code to get a -3 nanoseconds improvement is kind of an overkill. Especially as this is not such a hot path of code. For readability I'd prefer the original one. Otherwise: Good work!

Reply:
I am not sure that `char.ToString((char)(key - Key.D0 + '0'))` is more readable than `s_digits[key - Key.D0]`, but that is likely a matter of personal opinion.

Performance-wise I agree this probably doesn't matter, especially since this is Key to string path, so not used during XAML loading, typically it would be showing shortcuts in menu entries and UI and for that the performance will get drowned in much bigger costs.

@h3xds1nz Curious, what is the allocation in the Original case for? Are the strings not interned?

Reply:
@miloush `char.ToString` will forcefully init a new string instance everytime, there's no pool even for printable ASCII, that's where the 24B is coming from. Most characters are used solo as strings too (e.g. formats) throughout the assemblies, so the cost is really just mostly with the array itself. if you go that path.

I agree it is not a hot path and hence it was more like a point to consider; however I also think on the side of readibility the array solution makes more sense.

But in my own opinion about the implementation, I'd go with the `char` solution too, since the 3 ns doesn't matter here as you both pointed out. I like to discuss.

Reply:
> I am not sure that [...] is more readable than

You are right it's a matter of personal opinion. Thus I wouldn't have said something but h3xds1nz asked for it. But to explain my opinion: It's not the replacement of the lines, its the declaration of the two arrays. These are outside of the function. People who are reading the code have to figure out what they are for / where they are used etc. Do this a couple of times and the code becomes hard(er) to read. That's why I prefer to avoid it when it's not necessary.

Reply:
Thanks guys, we will not move this direction.